src/app.js

```diff
@@ -300,32 +300,24 @@ class Messenger extends EventEmitter {
       attachments
     } = message;
 
-    if (this.options.emitGreetings && this.greetings.test(text)) {
-      const firstName = session.profile && session.profile.first_name.trim() || '';
-      const surName = session.profile && session.profile.last_name.trim() || '';
-      const fullName = `${firstName} ${surName}`;
-
-      this.emit('text.greeting', new Response(this, {event, senderId, session, firstName, surName, fullName}));
-      return;
-    }
-
-    if (this.help.test(text)) {
-      this.emit('text.help', new Response(this, {event, senderId, session}));
+    if (this.emitOptionalEvents(event, senderId, session, text)) {
       return;
     }
 
     if (quickReply) {
-      debug('message.quickReply payload: "%s"', quickReply.payload);
+      const text = quickReply.payload;
+      const normalizedText = text.toLowerCase().trim();
+      debug('message.quickReply payload: "%s"', text);
 
-      this.emit('text', new Response(this, {event, senderId, session, source: 'quickReply', text: quickReply.payload}));
-      this.emit('message.quickReply', new Response(this, {event, senderId, session, payload: quickReply.payload}));
```

Comment:
ok, I just need to double check the nuances between `title` and `payload` (same for postback) https://developers.facebook.com/docs/messenger-platform/send-api-reference/quick-replies

I think the `.title` is what's shown in the message thread, which is technically the "text", while `.payload` is the only thing the webhook gets from Facebook. To accommodate users re-typing the title, `.title` and `.payload` should be the same thing; this is where `text: payload` came in. For the `text` event, all we had was the `.payload`, so as long as they were the same, it worked.

However, for `message.quickReply`, we don't get the text, we get the payload. If users of the sdk see that `text` comes in, I would confuse that with the button `title`, when what's actually sent is the button `payload`.

For quizbot, the quickreply title is "A" while the payload is `og1rix:0:a`. An event listener would expect `text: 'A'` and `payload: 'og1rix:0:a'`, but as the code is written now, they only get:
* `emit('text', {text: 'og1rix:0:a'})` -- ehhhh, kind of misleading, but trying to duck type `message.text` with what we got
* `emit('message.text', {text: 'og1rix:0:a'})` -- ehhhh, kind of misleading

but the way it was originally, the quickreply gets:
* `emit('text', {text: 'og1rix:0:a'})` -- ehhhh, kind of misleading, but trying to duck type `message.text` with what we got
* `emit('message.text', {payload: 'og1rix:0:a'})` -- right

So in this case, adding consistency comes at the cost of creating ambiguity